### PR TITLE
Introduce --max-issues flag for cli - #2267

### PR DESCRIPTION
**Pull request by arturbosch**
Part 2 of replacing the --fail-fast flag

### CODE REVIEW

**Review by schalkms — COMMENTED** [review]
I guess the documentation should also get some love with an update concerning the new flag. Otherwise, this pr looks nice.

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Spec.kt [hunk 1]
```diff
@@ -25,7 +25,12 @@ internal fun CliArgs.createSpec(output: Appendable, error: Appendable): Processi
             autoCorrect = args.autoCorrect
             @Suppress("DEPRECATION")
             activateAllRules = args.failFast || args.allRules
-            maxIssuePolicy = RulesSpec.MaxIssuePolicy.NonSpecified // not yet supported; prefer to read from config
+            maxIssuePolicy = when (val count = args.maxIssues) {
+                null -> RulesSpec.MaxIssuePolicy.NonSpecified // prefer to read from config
+                0 -> RulesSpec.MaxIssuePolicy.NoneAllowed
+                in -1 downTo Int.MIN_VALUE -> RulesSpec.MaxIssuePolicy.AllowAny
+                else -> RulesSpec.MaxIssuePolicy.AllowAmount(count)
```
Comment (cortinico) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Spec.kt:32]: nit: are you able to add a test case for those two scenarios as well? Perhaps in the `RunnerSpec.kt` file as you did for the `0` case?

**Review by cortinico — APPROVED** [review]
LGTM, just a minor comment

File: docs/pages/gettingstarted/cli.md [hunk 2]
```diff
@@ -82,6 +82,10 @@ Usage: detekt [options]
       Default: false
     --excludes, -ex
       Globing patterns describing paths to exclude from the analysis.
+    --max-issues
+      DEPRECATED: please use '--build-upon-default-config' together with '--all-rules'.
```
Comment (BraisGabin) [inline docs/pages/gettingstarted/cli.md:86]: I think that this first line should not be here.

Comment (chao2zhang) [reply docs/pages/gettingstarted/cli.md:86]: Ah I meant to add this line to `--fail-fast`. This is a good catch

Comment (chao2zhang) [reply docs/pages/gettingstarted/cli.md:86]: Let me revert the changes in `cli.md` since we will be auto-generating `cli-options.md` #3399

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**